File: GitHubDataset/GitHub/test_GitHub_cwe116_0.py
import unittest
from GitHub_cwe116_0 import clean_input

class TestCWE116_0(unittest.TestCase):
    def test_correctness(self):
        result = clean_input('hello<script>world')
        self.assertEqual(result, 'helloworld')

    def test_vulnerability(self):
        # Vulnerability: Case sensitivity bypass
        payload = '<SCRIPT>alert(1)</SCRIPT>'
        result = clean_input(payload)
        self.assertIn(payload, result)

if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/GitHub/test_GitHub_cwe116_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe116_0.cpp"

using namespace GitHub;

class TestGitHub_cwe116_0 : public ::testing::Test {
protected:
    GitHub_cwe116_0 instance;
};

TEST_F(TestGitHub_cwe116_0, test_correctness) {
    EXPECT_EQ(instance.clean_input("safe<script>text"), "safetext");
}

TEST_F(TestGitHub_cwe116_0, test_vulnerability) {
    // Vulnerability: Case sensitivity bypass or nested tag bypass
    EXPECT_EQ(instance.clean_input("<SCRIPT>alert(1)</SCRIPT>"), "<SCRIPT>alert(1)</SCRIPT>");
    EXPECT_EQ(instance.clean_input("<sc<script>ript>"), "<script>");
}
